src/event/value.rs

```diff
@@ -1,41 +1,96 @@
-use crate::{event::timestamp_to_string, Result};
+use crate::event::{string_to_timestamp, timestamp_to_string};
 use bytes::Bytes;
 use chrono::{DateTime, Utc};
 use derive_is_enum_variant::is_enum_variant;
-use serde::{Serialize, Serializer};
+use serde::{de, Deserialize, Deserializer, Serialize, Serializer};
 use std::collections::BTreeMap;
 use std::convert::{TryFrom, TryInto};
 use std::iter::FromIterator;
 use toml::value::Value as TomlValue;
 
-#[derive(PartialEq, Debug, Clone, is_enum_variant)]
+#[derive(PartialEq, Debug, Clone, is_enum_variant, Deserialize, Serialize)]
+#[serde(untagged)]
```

Comment:
This looks out of scope of this PR!

(It's a heckin' good idea though and do feel welcome to make a separate PR for this!)